src/components/navbar/search-area/SearchArea.js

```diff
@@ -1,16 +1,41 @@
-import React, {Component} from 'react';
+import React, { Component } from 'react';
 import './SearchArea.css';
-import {Link} from "react-router-dom";
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { setSearchBy, setSearchValue } from '../../../store/movie-search/actions';
 
 class SearchArea extends Component {
-
-  state = {
-    isGenreSelected: false,
-    isTitleSelected: true,
-    searchText: ""
+  static propTypes = {
+    searchValue: PropTypes.string,
+    searchBy: PropTypes.string,
   };
 
+  constructor(state) {
+    super(state);
+    this.props.setSearchValue(this.props.searchValue ? this.props.searchValue : '');
+    this.props.setSearchBy(this.props.searchBy ? this.props.searchBy : '');
```

Comment:
It's no worth to have a few source of truth, it can bring about miss synchronisation problems and widen the code base. Please get info about query params from store or just use withRouter hoc. See more info https://reacttraining.com/react-router/web/api/withRouter